test/FunctionalTests/Microsoft.OData.Core.Tests/Build.NetFramework/SelectExpandClauseExtensionsTests.cs

```diff
@@ -0,0 +1,138 @@
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.OData.Tests.UriParser;
+using Microsoft.OData.Tests.UriParser.Binders;
+using Microsoft.OData.UriParser;
+using Microsoft.OData.Edm;
+using Xunit;
+using ODataErrorStrings = Microsoft.OData.Strings;
+using Microsoft.OData.UriParser.Aggregation;
+
+namespace Microsoft.OData.Tests
+{
+    public class SelectExpandClauseExtensionsTests
+    {
+        [Fact]
+        public void GetCurrentLevelSelectListTestMultiSelect()
+        {
+            //Arrange
+            var expected = new List<string>();
+            expected.Add("Name");
+            expected.Add("Shoe");
+
+            //Act
+            var clause = CreateSelectExpandClauseMultiSelect();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.Equal(expected.Count, actual.Count);
+            Assert.Equal(0,expected.Except(actual).Count());
+            Assert.Equal(0,actual.Except(expected).Count());
+        }
+
+        [Fact]
+        public void GetCurrentLevelSelectListTestNestedSelect()
+        {
+            //Arrange
+            var expected = new List<string>();
+            expected.Add("Name");
+            expected.Add("FavoriteDate");
+            expected.Add("MyAddress/City");
+
+            //Act
+            var clause = CreateSelectExpandClauseNestedSelect();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.Equal(expected.Count, actual.Count);
+            Assert.Equal(0, expected.Except(actual).Count());
+            Assert.Equal(0, actual.Except(expected).Count());
+        }
+
+        [Fact]
+        public void GetCurrentLevelSelectListTestAllSelected()
+        {
+            //Arrange
+            var expected = new List<string>();
+            expected.Add("*");
+            
+            //Act
+            var clause = CreateSelectExpandClauseAllSelected();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.Equal(expected.Count, actual.Count);
+            Assert.Equal(expected[0], actual[0]);            
+        }
+
+        [Fact]
+        public void GetCurrentLevelSelectListTestAllSelectedWithNamespace()
+        {
+            //Arrange
+            var expected = new List<string>();
+            expected.Add("namespace.*");
+
+            //Act
+            var clause = CreateSelectExpandClauseAllSelectedWithNamespace();
+            var actual = SelectExpandClauseExtensions.GetCurrentLevelSelectList(clause);
+
+            //Assert
+            Assert.Equal(expected.Count, actual.Count);
+            Assert.Equal(expected[0], actual[0]);
+        }
+
+        private SelectExpandClause CreateSelectExpandClauseMultiSelect()

```

Comment:
It seems all this private method is only used once, why don't write the codes in the test case?

meanwhile, it seems these private method can be static

Reply:
Made it separate considering we might need to add more tests in the future and call these method from those tests.  Made it static
#Resolved